frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/ReleasePlan.tsx

```diff
@@ -0,0 +1,193 @@
+    const { refetch } = useReleasePlans(projectId, featureName, environment);
+    const { removeReleasePlanFromFeature, startReleasePlanMilestone } =
+        useReleasePlansApi();
+    const { setToastData, setToastApiError } = useToast();
+
+    const [removeOpen, setRemoveOpen] = useState(false);
+
+    const onRemoveConfirm = async () => {
+        try {
+            await removeReleasePlanFromFeature(
+                projectId,
+                featureName,
+                environment,
+                id,
+            );
+            setToastData({
+                title: `Release plan "${name}" has been removed from ${featureName} in ${environment}`,
+                type: 'success',
+            });
+            refetch();
+            setRemoveOpen(false);
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+    };
+
+    const onStartMilestone = async (milestone: TReleasePlanMilestone) => {
+        try {
+            await startReleasePlanMilestone(
+                projectId,
+                featureName,
+                environment,
+                id,
+                milestone.id,
+            );
+            setToastData({
+                title: `Milestone "${milestone.name}" has started`,
+                type: 'success',
+            });
+            refetch();
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+    };
+
+    const disabled = !activeMilestoneId;
+    const activeIndex = milestones.findIndex(
+        (milestone) => milestone.id === activeMilestoneId,
+    );
+
+    return (
+        <StyledContainer disabled={disabled}>
+            <StyledHeader disabled={disabled}>
+                <StyledHeaderTitleContainer>
+                    <StyledHeaderTitleLabel>
+                        Release plan
+                    </StyledHeaderTitleLabel>
+                    <span>{name}</span>
+                    <StyledHeaderDescription>
+                        {description}
+                    </StyledHeaderDescription>
+                </StyledHeaderTitleContainer>
+                <PermissionIconButton
+                    onClick={() => setRemoveOpen(true)}
+                    permission={DELETE_FEATURE_STRATEGY}
+                    environmentId={environment}
+                    projectId={projectId}
+                    tooltipProps={{
+                        title: 'Remove release plan',
+                    }}
+                >
+                    <Delete />
+                </PermissionIconButton>
+            </StyledHeader>
+            <StyledBody>
+                {milestones.map((milestone, index) => (
+                    <div key={milestone.id}>
+                        <ReleasePlanMilestone
+                            milestone={milestone}
+                            status={
+                                index === activeIndex
```

Comment:
Don't we have activeMilestoneId we could use here?

Reply:
We're already [using it](https://github.com/Unleash/unleash/pull/8843/files#diff-498111a84758346c8dba958921663db58596ddb5fdc5082da8b88f7173ea26d7R136-R138) here, even if it's indirectly, so in the end it shouldn't matter much. However I'm on board if you think this is more readable: https://github.com/Unleash/unleash/pull/8843/commits/f42b7883db3d8aaa725cc64c40ae06721c0cd043